From cca98c2c6ac3494f87202c745209734a493681d8 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Tue, 11 Mar 2014 11:16:32 +0800
Subject: [PATCH 295/310] ENGR00293990 Selinux:change rules for wifi tethering

For our wifi tethering is directly using wlan0 net interface which is different
from others which are based on virtual net interface.
So for selinux, we need to change rules to fix some denied issue for netd self
or with unlabeled process.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 evk_6sl/BoardConfig.mk        | 3 ++-
 evk_6sl/sepolicy/netd.te      | 3 +++
 sabreauto_6q/BoardConfig.mk   | 3 ++-
 sabreauto_6q/sepolicy/netd.te | 3 +++
 sabresd_6dq/BoardConfig.mk    | 3 ++-
 sabresd_6dq/sepolicy/netd.te  | 3 +++
 6 files changed, 15 insertions(+), 3 deletions(-)
 create mode 100644 evk_6sl/sepolicy/netd.te
 create mode 100644 sabreauto_6q/sepolicy/netd.te
 create mode 100644 sabresd_6dq/sepolicy/netd.te

diff --git a/evk_6sl/BoardConfig.mk b/evk_6sl/BoardConfig.mk
index 1e06d1b..eb80e5e 100644
--- a/evk_6sl/BoardConfig.mk
+++ b/evk_6sl/BoardConfig.mk
@@ -82,4 +82,5 @@ TARGET_BOOTLOADER_CONFIG := mx6sl:mx6sl_evk_android_config
 BOARD_SEPOLICY_DIRS := \
        device/fsl/evk_6sl/sepolicy
 BOARD_SEPOLICY_UNION := \
-       app.te
+       app.te \
+       netd.te
diff --git a/evk_6sl/sepolicy/netd.te b/evk_6sl/sepolicy/netd.te
new file mode 100644
index 0000000..80966f5
--- /dev/null
+++ b/evk_6sl/sepolicy/netd.te
@@ -0,0 +1,3 @@
+allow netd self:netlink_socket *;
+allow netd self:packet_socket *;
+allow netd unlabeled:sock_file { create_file_perms rwx_file_perms relabelfrom };
diff --git a/sabreauto_6q/BoardConfig.mk b/sabreauto_6q/BoardConfig.mk
index 2fbcaa2..9f2acd8 100644
--- a/sabreauto_6q/BoardConfig.mk
+++ b/sabreauto_6q/BoardConfig.mk
@@ -107,4 +107,5 @@ BOARD_SEPOLICY_DIRS := \
        device/fsl/sabreauto_6q/sepolicy
 
 BOARD_SEPOLICY_UNION := \
-       app.te
+       app.te \
+       netd.te
diff --git a/sabreauto_6q/sepolicy/netd.te b/sabreauto_6q/sepolicy/netd.te
new file mode 100644
index 0000000..80966f5
--- /dev/null
+++ b/sabreauto_6q/sepolicy/netd.te
@@ -0,0 +1,3 @@
+allow netd self:netlink_socket *;
+allow netd self:packet_socket *;
+allow netd unlabeled:sock_file { create_file_perms rwx_file_perms relabelfrom };
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index 8f4b48e..5c3245d 100755
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -113,4 +113,5 @@ BOARD_SEPOLICY_DIRS := \
        device/fsl/sabresd_6dq/sepolicy
 
 BOARD_SEPOLICY_UNION := \
-       app.te
+       app.te \
+       netd.te
diff --git a/sabresd_6dq/sepolicy/netd.te b/sabresd_6dq/sepolicy/netd.te
new file mode 100644
index 0000000..80966f5
--- /dev/null
+++ b/sabresd_6dq/sepolicy/netd.te
@@ -0,0 +1,3 @@
+allow netd self:netlink_socket *;
+allow netd self:packet_socket *;
+allow netd unlabeled:sock_file { create_file_perms rwx_file_perms relabelfrom };
-- 
1.8.0

